### PR TITLE
ClientAI - Add Unitary Tests

### PR DESCRIPTION
**Pull request by 31Nathan**
Add unitary tests for the AI objects:
The Map object is tested
The Inventory object is tested
The ServerWrapper object is tested
The DLLlibWrapper object is tested

### CODE REVIEW

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Review by GuyomT — APPROVED** [review]
NTR

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
- Why is there a .so inside the source files ?
- Why are you not using a testing framework ?
- How to check the coverage of the tests ?
- How to launch the tests ?

**Comment by 31Nathan** [comment]
> * Why is there a .so inside the source files ?
> * Why are you not using a testing framework ?
> * How to check the coverage of the tests ?
> * How to launch the tests ?

The .so can be deleted, its presence is only understandable because it wanted to test the object with as much lib as i can
Because the assert works well too
For the coverage I don't know
To launch the tests, you go in the zappy_ai_tests directory and you can execute each one of the tests
If necessary, I can make a bash script in the test dir at the root directory to launch them all

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
- Remove the TestLib.so or put it in an other directory (inside the tests/lib folder maybe), this doesn't fit the coding style (major error) 
- Move all of the tests file inside the tests folder as well
- Assert is not a testing framework, even though it works properly
- Why not launching the tests at the same time as the other tests ? 
    - https://stackoverflow.com/questions/52494425/running-a-pytest-test-from-cmake-where-the-test-and-sources-are-in-different-fol

**Comment by 31Nathan** [comment]
> * Remove the TestLib.so or put it in an other directory (inside the tests/lib folder maybe), this doesn't fit the coding style (major error)
> * Move all of the tests file inside the tests folder as well
> * Assert is not a testing framework, even though it works properly
> * Why not launching the tests at the same time as the other tests ?
>   
>   * https://stackoverflow.com/questions/52494425/running-a-pytest-test-from-cmake-where-the-test-and-sources-are-in-different-fol

For the .so okay
Moving the tests will be changing the code lines "sys.path.append" and it will be even dirtier than the actual, I can make symlink of tests if necessary (Already discussed with @LucasTesnier)
I can launch them like that if you want

File: tests/zappy_ai_tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+find_package(Python COMPONENTS Interpreter REQUIRED)
+
+add_test(NAME zappy_ai_unit_test
+    COMMAND ${Python_EXECUTABLE} -m pytest ${CMAKE_CURRENT_SOURCE_DIR}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/src/zappy_ai_src
+)
```
Comment (Cavonstavant) [inline tests/zappy_ai_tests/CMakeLists.txt:6]: :+1: :+1: :+1: :+1: :+1:

**Review by Cavonstavant — APPROVED** [review]
VAMOS!

**Pull request merged**